File: veloProWeb/src/main/java/com/veloProWeb/Validation/CostumerValidator.java
package com.veloProWeb.Validation;

import com.veloProWeb.Model.Entity.Costumer.Costumer;
import org.springframework.stereotype.Component;

@Component
public class CostumerValidator {
    public void validate(Costumer costumer){
        validateName(costumer.getName());
        validateSurname(costumer.getSurname());
        validatePhone(costumer.getPhone());
        validateEmail(costumer.getEmail());
    }
    public void validateValuePayment(int number, Costumer costumer){
        if (number <= 0){
            throw new IllegalArgumentException("El monto no puede ser menor a 0.");
        }
        if (number > costumer.getDebt()){
            throw new IllegalArgumentException("El monto supera el valor de la deuda.");
        }
    }

    private void validateName(String name){
        if (name == null || name.trim().isBlank() || name.trim().length() < 3 || !name.matches("[a-zA-Z ]+")){
            throw new IllegalArgumentException("Ingrese nombre válido.");
        }
    }
    private void validateSurname(String surname){
        if (surname == null || surname.trim().isBlank() || surname.trim().length() < 3 || !surname.matches("[a-zA-Z ]+")){
            throw new IllegalArgumentException("Ingrese apellido válido.");
        }
    }
    private void validatePhone(String phone){
        if (phone.trim().isBlank() || phone.trim().length() != 13){
            throw new IllegalArgumentException("Ingrese número válido, Ej: +569 12345678");
        }
    }
    private void validateEmail(String email){
        if (email.trim().isBlank() || !email.matches("[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}")){
            throw new IllegalArgumentException("Ingrese Email válido.");
        }
    }
}

File: veloProweb-frontend/src/app/validation/costumer-validator.ts
import { Costumer } from "../models/Costumer/costumer.model";

export class CostumerValidator {
    static validateForm(selectedCostumer: Costumer): boolean {
        return this.isFieldValid(selectedCostumer, 'name') &&
            this.isFieldValid(selectedCostumer, 'surname') &&
            this.isFieldValid(selectedCostumer, 'phone') &&
            this.isFieldValid(selectedCostumer, 'email');
    }

    static isFieldValid(costumer: Costumer, fieldName: keyof Costumer): boolean {
        const value = costumer[fieldName];
        if (fieldName === 'email') {
            return !value || typeof value === 'string' && /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/.test(value);
        }
        if (typeof value !== 'string' || value.trim().length < 3) {
            return false;
        }
        if (fieldName === 'phone' && value.trim().length < 13) {
            return false;
        }
        if (fieldName === 'name' || fieldName === 'surname') {
            return /^[a-zA-Z ]+$/.test(value.trim());
        }
        return true;
    }
}
